File: AndroidRCLNested/app/src/main/java/com/example/shakil/androidrclnested/Adapter/MyAdapter.java
package com.example.shakil.androidrclnested.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.shakil.androidrclnested.Model.Model;
import com.example.shakil.androidrclnested.R;
import com.squareup.picasso.Picasso;

import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    Context context;
    List<Model> modelList;

    public MyAdapter(Context context, List<Model> modelList) {
        this.context = context;
        this.modelList = modelList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(context).inflate(R.layout.view_holder_item, viewGroup, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int i) {
        Picasso.get().load(modelList.get(i).getImage_link()).into(myViewHolder.image_view);
        myViewHolder.text_view.setText(modelList.get(i).getText());
    }

    @Override
    public int getItemCount() {
        return modelList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {

        ImageView image_view;
        TextView text_view;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            image_view = itemView.findViewById(R.id.image_view);
            text_view = itemView.findViewById(R.id.text_view);
        }
    }
}

File: KotlinRCLNested/app/src/main/java/com/example/shakil/kotlinrclnested/Adapter/MyAdapter.kt
package com.example.shakil.kotlinrclnested.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.example.shakil.kotlinrclnested.Model.MyModel
import com.example.shakil.kotlinrclnested.R
import com.squareup.picasso.Picasso

class MyAdapter(private val context: Context, private val modelList: List<MyModel>): RecyclerView.Adapter<MyAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): MyViewHolder {
        val itemView = LayoutInflater.from(context).inflate(R.layout.layout_item, p0, false)
        return MyViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return modelList.size
    }

    override fun onBindViewHolder(p0: MyViewHolder, p1: Int) {
        Picasso.get().load(modelList[p1].image).into(p0.imageView)
        p0.textView.text = modelList[p1].text
    }

    class MyViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
        internal var imageView: ImageView
        internal var textView: TextView

        init {
            imageView = itemView.findViewById(R.id.image_view) as ImageView
            textView = itemView.findViewById(R.id.text_view) as TextView
        }
    }
}